aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/scm/GitStepTest.java

```diff
@@ -162,4 +166,28 @@
         assertFalse(iterator.hasNext());
     }
 
+    @Ignore("This test is currently disabled until the fix for JENKINS-29326 is released in the git plugin 2.4.1 or later.")
+    @Issue("JENKINS-29326")
+    @Test
+    public void identicalGitSCMs() throws Exception {
+        otherRepo.git("init");
+        otherRepo.write("otherfile", "");
+        otherRepo.git("add", "otherfile");
+        otherRepo.git("commit", "--message=init");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "demo");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "    ws {\n" +
+                        "        dir('main') {\n" +
+                        "            git($/" + otherRepo + "/$)\n" +
+                        "        }\n" +
+                        "        dir('other') {\n" +
+                        "            git($/" + otherRepo + "/$)\n" +
+                        "        }\n" +
+                        "        archive '**'\n" +
+                        "    }\n" +
+                        "}"));
+        WorkflowRun b = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        assertEquals(1, b.getActions(BuildData.class).size());
```

Comment:
And you need to verify that `p.getSCMs()` has only entry. While `WorkflowRun.checkouts` may have duplicates, since we have no way of detecting them, `getSCMs` _can_, by comparing `SCM.getKey`.

That would allow https://github.com/jenkinsci/bitbucket-plugin/pull/20 to be reverted.